contracts/teller-network/Escrow.sol

```diff
@@ -218,327 +188,256 @@ contract Escrow is Pausable, MessageSigned, Fees, Arbitrable, RelayRecipient {
-     *         Transaction must not be canceled or released
-     */
-    function withdraw_emergency(uint _escrowId) external whenPaused {
-        EscrowTransaction storage trx = transactions[_escrowId];
-        require(trx.status == EscrowStatus.FUNDED, "Cannot withdraw from escrow in a stage different from FUNDED. Open a case");
-
-        address payable seller = metadataStore.getOfferOwner(trx.offerId);
-        _cancel(_escrowId, seller, trx);
-    }
+        status = EscrowStatus.CANCELED;
 
-    /**
-    * @dev Fallback function
-    */
-    function() external {
+        emit Canceled();
     }
 
     /**
      * @dev Rates a transaction
-     * @param _escrowId Id of the escrow
      * @param _rate rating of the transaction from 1 to 5
      * @notice Requires contract to not be paused.
      *         Can only be executed by the buyer
      *         Transaction must released
      */
-    function rateTransaction(uint _escrowId, uint _rate) external whenNotPaused {
+    function rateTransaction(uint _rate) external {
         require(_rate >= 1, "Rating needs to be at least 1");
         require(_rate <= 5, "Rating needs to be at less than or equal to 5");
-        require(!isDisputed(_escrowId), "Can't rate a transaction that has an arbitration process");
-
-        EscrowTransaction storage trx = transactions[_escrowId];
+        require(!arbitrations.isDisputed(address(this)), "Can't rate a transaction that has an arbitration process");
 
-        require(trx.rating == 0, "Transaction already rated");
-        require(trx.status == EscrowStatus.RELEASED, "Transaction not released yet");
-        require(trx.buyer == get_sender(), "Function can only be invoked by the escrow buyer");
+        EscrowManagement mgmt = EscrowManagement(factory);
 
-        trx.rating = _rate;
+        require(mgmt.rating(address(this)) == 0, "Transaction already rated");
+        require(status == EscrowStatus.RELEASED, "Transaction not released yet");
+        require(buyer == msg.sender, "Function can only be invoked by the escrow buyer");
 
-        emit Rating(trx.offerId, trx.buyer, _escrowId, _rate, block.timestamp);
+        mgmt.rate(seller, buyer, _rate);
     }
 
-
     /**
      * @notice Open case as a buyer or seller for arbitration
-     * @param _escrowId Id of the escrow
-     * @dev Consider using Aragon Court for this.
      */
-    function openCase(uint _escrowId, string calldata motive) external {
-        EscrowTransaction storage trx = transactions[_escrowId];
-
-        require(!isDisputed(_escrowId), "Case already exist");
-        require(trx.buyer == get_sender() || metadataStore.getOfferOwner(trx.offerId) == get_sender(), "Only a buyer or seller can open a case");
-        require(trx.status == EscrowStatus.PAID, "Cases can only be open for paid transactions");
+    function openCase(string calldata _motive) external {
+        require(buyer == msg.sender || seller == msg.sender, "Only a buyer or seller can open a case");
```

Comment:
Should we create a modifier for this, we reuse it a lot, maybe might save some gas to deploy the contract

Reply:
Good idea!